### ISSUE TITLE
feat(abi): asyncify KVBackend.Prefill — add residency-transfer methods so a remote L3 KV backend can attach

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#638]
_Shared-spine seam edit for the throughput-trust portfolio (step S6a). Unblocks a remote/disaggregated KV backend for #53 (agentl3)._

## Context

The KV-MMU enforcement seam was inverted to a per-session factory (`abi.RegisterKVBackend` / `kvmmu.NewBackend`), so `kvmmu` already enforces a quarantine verdict against **any** registered `abi.KVBackend` with zero concrete-`model` dependency (`internal/kvmmu/kvmmu.go:131`). That is the seam a remote/disaggregated L3 KV backend attaches behind — the same way the region/page-out backends do, no edit to the `kvmmu` composer.

**The one place the seam is still single-node-shaped (verified):** `abi.KVBackend.Prefill` returns a **synchronous dense `[]float32`** logits vector:

```go
// internal/abi/registry.go:635
Prefill(ids []int) []float32 // prefill a span; return next-token logits
```

inherited from the in-process `model.Session.Prefill` (`internal/model/kv.go:386`). A network-backed L3 KV backend cannot return dense logits at line rate — it implies a round-trip per call, and the signature has **no async path, no batching, no error channel, no `ctx`**. So a remote backend cannot satisfy this seam as-is; the disaggregated direction stays a stub until the shape is widened. `[CONFIRMED]` by the shared-spine verification pass.

## Scope

Add an **additive** residency-transfer method set to the `abi.KVBackend` seam, shaped for a span fak does **not** host locally, without breaking the in-process consumer:

1. A **stage/restore-by-digest** method pair returning a **typed outcome** (a `cachemeta.KVTransferVerdict`-shaped `ok | MISS | FAULT`, never a silent recompute), **not** a dense logits vector — so the `kvmmu` enforcement path can target a span on a remote tier.
2. Keep the in-process adapter (`internal/model/kvbackend.go`, `kvBackend{s *Session}`) **byte-identical**: the new methods default to the local synchronous path; `Prefill`'s existing behavior is unchanged for the `inkernel` engine.
3. The seam edit must compose with the frozen ABI re-export (`pkg/abi`) — a vendor driver implementing the widened `KVBackend` is still `internal/abi.KVBackend` (no ABI break).
4. Thread an error/`ctx` channel so a remote stall is a typed failure, not a hang.

## Acceptance

- [ ] `abi.KVBackend` carries residency-transfer methods returning a typed outcome (not dense logits); a fake remote backend exercises them in `internal/kvmmu/kvbackend_seam_test.go`-style.
- [ ] The in-process `model` adapter is byte-identical on the existing `Prefill`/`Evict`/`Len`/`ModelID` path (existing kvmmu + modelengine tests stay green, bit-for-bit).
- [ ] No frozen-ABI break: `pkg/abi` re-export compiles; an out-of-tree driver can register a backend implementing the widened interface.
- [ ] A remote/stub L3 backend can be registered via `abi.RegisterKVBackend` and `kvmmu.NewBackend` enforces an eviction against an L3-resident span **without** any concrete-`model` dependency.

## Honest fences

- This is the **seam edit only** — it unblocks a remote L3 KV backend; it does **not** ship the byte-moving transport (that is #29 / `StageTransport`) or the `L3RegionBackend` impl (#53 child-B).
- Bit-exact middle-span `Evict` still requires fak to own the pre-RoPE `Kraw` rows (`internal/model/kvcache.go:128`); the residency-transfer methods move fak-serialized spans (so the moat survives off-box) — they do **not** make exact-span eviction work against an opaque-page remote engine that owns the KV.
- Recurrent/hybrid caches still reject mid-span evict (`RecurrentEvictUnsupportedError`); the widened seam does not change that.

**Comment by anthony-chaudhary** [comment]
Parent: #637 (shared-spine sequencing, step S6a). This is the seam edit only — it unblocks a remote L3 KV backend; the byte-moving transport is #29 and the `L3RegionBackend` impl is #53 child-B.

**Comment by anthony-chaudhary** [comment]
Resolved by `4cae2b9` (resolving commit). Closed by the DOS dispatch loop's close-resolved arm, witnessed via `dos commit-audit` (verdict OK / diff-witnessed). Reopen if this does not fully resolve it.

**Comment by anthony-chaudhary** [comment]
Shipped in `4cae2b9` (on `origin/main`), `feat(abi): add async residency-transfer methods to the KVBackend seam (#638) (fak abi)`.

The seam was widened additively — `abi.KVBackend` now carries `StageSpan` / `RestoreSpan`, a stage/restore-by-digest pair returning a typed `KVResidency` (`ok | MISS | FAULT`) with a `ctx`, instead of the dense `[]float32` logits `Prefill` returns. A network-backed L3 KV backend can now satisfy the seam: a typed MISS/FAULT is reported (never a silent recompute), and a remote stall surfaces as a typed FAULT rather than a hang.

Acceptance:

- [x] **`abi.KVBackend` carries residency-transfer methods returning a typed outcome (not dense logits); a fake remote backend exercises them.** `StageSpan`/`RestoreSpan` → `abi.KVResidency` (`internal/abi/kvbackend.go`, `internal/abi/registry.go`). `internal/kvmmu/kvbackend_seam_test.go::TestRemoteL3BackendEnforcedWithoutModelDependency` (the `stubL3` pure-remote backend) exercises the pair.
- [x] **In-process `model` adapter byte-identical on the existing `Prefill`/`Evict`/`Len`/`ModelID` path.** Only the two new methods were added (`internal/model/kvbackend.go`); the existing `internal/model` + `internal/kvmmu` + `internal/modelengine` tests stay green. The local default is the synchronous path — `StageSpan` = no-op OK, `RestoreSpan` = typed MISS (no off-box tier).
- [x] **No frozen-ABI break: `pkg/abi` re-export compiles; an out-of-tree driver can implement the widened interface.** `pkg/abi/abi.go` re-exports `KVResidency` / `KVResidencyOutcome` + the constants; `pkg/abi` builds and tests green. The new types live in `internal/abi` (which imports nothing internal), so the seam stays cycle-free — `cachemeta` imports `abi`, never the reverse.
- [x] **A remote/stub L3 backend registered via `abi.RegisterKVBackend`, enforced by `kvmmu.NewBackend` against an L3-resident span with zero concrete-`model` dependency.** `TestRemoteL3BackendEnforcedWithoutModelDependency` registers the pure stub through `abi.RegisterKVBackend`, enforces a quarantine eviction through the pure-abi `NewBackendWithGate` seam (no `*model.Session`), and restores the in-process default after.

Bonus: `internal/cachemeta/kvresidency.go::FromKVResidency` bridges the outcome onto the same `LookupVerdict` trichotomy `KVTransferVerdict` produces, so a remote restore flows into the cache-meta lookup path identically to an in-stream `kv_transfer` entry.

**Honest fences (unchanged):** seam edit only — does **not** ship the byte-moving transport (#29 / `StageTransport`) or the `L3RegionBackend` impl (#53 child-B). Bit-exact middle-span `Evict` still requires fak to own the pre-RoPE `Kraw` rows; the residency pair moves fak-serialized spans so the moat survives off-box.
